python-mini-projects/projects/Get_meta_information_of_images/gps_utils.py

```diff
@@ -0,0 +1,23 @@
+import exifread
+import requests
+from geopy.geocoders import Nominatim
+
+def format_lati_long(data):
+	list_tmp=str(data).replace('[', '').replace(']', '').split(',')
+	list=[ele.strip() for ele in list_tmp]
+	if (list[-1].find('/') != -1):
+		data_sec = int(list[-1].split('/')[0]) /(int(list[-1].split('/')[1])*3600)
+	else:
+		data_sec = int(list[-1])/3600
+	data_minute = int(list[1])/60
+	data_degree = int(list[0])
+	result=data_degree + data_minute + data_sec
+	return result
+
+def get_location(filename):
+    img=exifread.process_file(open(filename,'rb'))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).